--- gd-2.0.17/gdft.c.orig	2003-12-24 22:42:13.000000000 +0100
+++ gd-2.0.17/gdft.c	2003-12-26 17:16:06.413285048 +0100
@@ -820,6 +820,9 @@
     }
 }
 
+/* for binary compatibility with gd 2.0.15- (called by e.g. php) */
+void gdFreeFontCache() __attribute__((alias("gdFontCacheShutdown")));
+	
 /********************************************************************/
 /* gdImageStringFT -  render a utf8 string onto a gd image          */
 
