--- gd-2.0.9/gd.h.orig	Wed Dec 11 17:37:14 2002
+++ gd-2.0.9/gd.h	Fri Jan 10 21:03:16 2003
@@ -7,7 +7,7 @@
 #endif
 
 /* default fontpath for unix systems  - whatever happened to standards ! */
-#define DEFAULT_FONTPATH "/usr/X11R6/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/truetype:/usr/X11R6/lib/X11/fonts/TTF:/usr/share/fonts/TrueType:/usr/share/fonts/truetype:/usr/openwin/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/Type1"
+#define DEFAULT_FONTPATH "/usr/share/fonts/TTF:/usr/share/fonts/Type1"
 #define PATHSEPARATOR ":"
 
 /* gd.h: declarations file for the graphic-draw module.
