--- gd-2.0.12/gd.h.wiget	Tue Apr 15 11:48:57 2003
+++ gd-2.0.12/gd.h	Tue Apr 15 11:49:52 2003
@@ -9,7 +9,7 @@
 /* some might want to set DEFAULT_FONTPATH from configure in config.h */
 #ifndef DEFAULT_FONTPATH
 /* default fontpath for unix systems  - whatever happened to standards ! */
-#define DEFAULT_FONTPATH "/usr/X11R6/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/truetype:/usr/X11R6/lib/X11/fonts/TTF:/usr/share/fonts/TrueType:/usr/share/fonts/truetype:/usr/openwin/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/Type1"
+#define DEFAULT_FONTPATH "/usr/share/fonts/TTF:/usr/share/fonts/Type1"
 #endif
 #ifndef PATHSEPARATOR
 #define PATHSEPARATOR ":"
