--- libgd-2.2.3/src/gd_topal.c.orig	2016-07-21 10:06:42.000000000 +0200
+++ libgd-2.2.3/src/gd_topal.c	2016-08-27 21:45:24.911652643 +0200
@@ -1545,7 +1545,7 @@
 #ifdef HAVE_LIBIMAGEQUANT
 	if (oim->paletteQuantizationMethod == GD_QUANT_DEFAULT ||
 	        oim->paletteQuantizationMethod == GD_QUANT_LIQ) {
-		liq_attr *attr = liq_attr_create_with_allocator(gdMalloc, gdFree);
+		liq_attr *attr = liq_attr_create(); // not passing gdMalloc/gdFree as allocators: they are just malloc/free, but liq requires 16-byte alignment
 		liq_image *image;
 		liq_result *remap;
 		int remapped_ok = 0;
